src/index.ts

```diff
@@ -361,6 +361,24 @@ export class ProjectConfig {
 
     return true;
   }
+
+  /**
+   * Generate a JSON string serialization of this configuration. File paths
+   * will be relative to root.
+   */
+  toJSON(): string {
+    const relative = (p: string) => p ? path.relative(this.root, p) : undefined;
```

Comment:
Are the types right here? If so, I think this should be more explicit, like:

```typescript
(p: string) => p !== '' ? path.relative(this.root, p) : undefined
```

If you meant to check for nulls here, hm... actually, is null checking turned on in this repo?